### PR TITLE
Add -batch flag to libndt-client

### PR DESCRIPTION
**Pull request by robertodauria**
A first attempt to add a `-batch` flag so we can embed libndt-client into Murakami.

The rationale behind it is to have a way to make libndt-client print JSON only, so that its output can be easily parsed and stored by Murakami.

@bassosimone this is likely incomplete:
- `-batch` does something only when performing an ndt7 download test 
- I'm not sure what the JSON format for the upload test should be / how to get the BBR data
- Does printing the JSON to stdout (while everything else is printed on stderr) seem a good solution to you? This would allow murakami to invoke the client like `./libndt-client -ndt7 -download -batch 2> /dev/null`.

### CODE REVIEW

**Comment by bassosimone** [comment]
@robertodauria:

> -batch does something only when performing an ndt7 download test

I guess it's fine.

> I'm not sure what the JSON format for the upload test should be / how to get the BBR data

This can perhaps be done at a later time. It involves getting the file descriptor, which can easily do (yay!), and calling `TCP_INFO` and `TCP_CC_INFO`. See:

1. https://github.com/m-lab/ndt-server/blob/master/tcpinfox/tcpinfox_linux.go

2. https://github.com/m-lab/ndt-server/blob/master/bbr/bbr_linux.go

As regards the data format, see https://github.com/m-lab/ndt-server/blob/master/spec/ndt7-protocol.md#measurement-message

We've changed the data format, but this client is basic enough that it does not seem to care, as far as I can tell.

> Does printing the JSON to stdout (while everything else is printed on stderr) seem a good solution to you? This would allow murakami to invoke the client like ./libndt-client -ndt7 -download -batch 2> /dev/null

Yes, this seems to me the right thing to do.

**Comment by bassosimone** [comment]
One more thing, do you understand what's wrong with AppVeyor?

File: libndt.hpp [hunk 1]
```diff
@@ -1922,10 +1926,14 @@ bool Client::ndt7_download() noexcept {
...
-      // is not going to be a real probem, it's just a theoric issue.
+      // is not going to be a real problem, it's just a theoric issue.
       if (count <= SIZE_MAX) {
         std::string sinfo{(const char *)buff.get(), (size_t)count};
-        on_result("ndt7", "download", std::move(sinfo));
+        if (settings_.batch_mode) {
+          std::cout << sinfo;
+        } else {
+          on_result("ndt7", "download", std::move(sinfo));
+        }
```
Comment (bassosimone) [inline libndt.hpp]: Can you maybe override `on_result` in the main client and move this logic there?

**Comment by robertodauria** [comment]
> One more thing, do you understand what's wrong with AppVeyor?

```
Build exceeded allowed resource quotas. Fix your build to consume less resources or contact AppVeyor support to request quotas increase.
```

Looks like the build took too long?

**Comment by bassosimone** [comment]
@robertodauria That would surprise me because usually builds are fast. But, I am instead wondering about a bandwidth quota? (We probably send and receive very fast with AppVeyor)

If that is the case, I am wondering how we can throttle the build. With Travis this is easy to do with Docker and makes me happy because we can make it see a Cable connection. How to do this with a Windows system is beyond my current depth.

**Pull request merged**